### ISSUE TITLE
Cannot set up forward search with Zathura (Texlab)

### ISSUE THREAD

**Issue by coxackie** [neovim/nvim-lspconfig#494]
First of all, let me say that you are doing a brilliant job!

 I am trying to set up forward search for tex and zathura using texlab, and instructions from https://texlab.netlify.app/docs/reference/configuration. (It is mentioned to check this out in the instructions about texlab nvim-lsp config).

I have the following in my `init.vim`:

```
lua << EOF
local lspconfig = require'lspconfig'
lspconfig.texlab.setup{
  settings = {
    latex = {
      build = {
        onSave = true;
      },
      forwardSearch = {
        executable = "zathura",
        args = {"--synctex-forward", "%l:1:%f", "%p"},
        onSave = true;
      }
    }
  }
}
EOF
```
Build-on-save works, but I do not get forward synctex. Zathura does not even start. Checkhealth shows no problems, and I get no complaints, just nothing happening in terms of forward search. Using `NVIM v0.5.0-dev+950-g5ce328df4`. Any ideas?

**Comment by mjlbach** [comment]
Can you paste a log? Add vim.lsp.set_log_level("debug") your the lua portion of your config and upload $HOME/.local/share/nvim/lsp.log.

**Comment by coxackie** [comment]
Sure - take a look at the attached.
[lsp.log](https://github.com/neovim/nvim-lspconfig/files/5743921/lsp.log)

**Comment by mjlbach** [comment]
There aren't any errors, I'm wondering if this is a client-side deficit (as in, it specifies we need to support SyncTeX?). Can you explain briefly what forward search is and what is currently missing? I use emacs for LaTeX (and have used a vimtex in the past), so I'm not super familiar with the built-in capabilities. Can you provide a minimal init.vim and tex document?

**Comment by coxackie** [comment]
Forward search should be able to take you to the part of the pdf that your cursor is when you save/build your tex file. This is the purpose of synctex. In principle it can also do backwards search, double clicking in the pdf should take you to the file, but it is more complicated.

 You can use any plain vanilla init vim and tex file - just put the lsp support (as I have) and try to test it. Just for info: [vimtex](https://github.com/lervag/vimtex) is able to do both backwards and forward search. The page at https://texlab.netlify.app/docs/reference/configuration shows how to implement this (I followed the ideas), but I guess it pertains to vs code (seeing how it is written)...

**Comment by coxackie** [comment]
ok, I have changed my setting a bit to:
```
local lspconfig = require'lspconfig'
vim.lsp.set_log_level("debug")
lspconfig.texlab.setup{
  settings = {
    latex = {
      rootDirectory = ".",
      build = {
        args = { "-pdf", "-interaction=nonstopmode", "-synctex=1", "-pvc" },
        forwardSearchAfter = true,
        onSave = true
      },
      forwardSearch = {
        executable = "zathura",
        args = {"--synctex-forward", "%l:1:%f", "%p"},
        onSave = true
      }
    }
  }
}
```
This now opens the pdf upon build, but I cannot do anything about forward Search. But I did notice that only the function
 - TexlabBuild: Build the current buffer
is implemented. There are more functions implemented in the description at the Texlab page. I guess ForwardSearch has not been implemented yet? Who should I reach out to for this?

**Comment by mjlbach** [comment]
There's an open PR implementing forward search https://github.com/neovim/nvim-lspconfig/pull/420

**Comment by coxackie** [comment]
ok, thanks, I have put a comment there to hopefully get some attention.

**Comment by coxackie** [comment]
right, so with the latest nightly (v0.5.0-dev+975-ga58c5509d) there is not even plain building with `TexlabBuild`. A "Build success" returns, but nothing happens.

**Comment by mjlbach** [comment]
I would file another issue, as it's unrelated to this one, but I don't see any changes that would affect it. In general I would recommend checking the log, and then doing a git bisect to nail down which commit caused the issue.

**Comment by coxackie** [comment]
hmm, ok, but is there noone else using this? Have you tried it yourself?

**Comment by mjlbach** [comment]
I'm going to close this issue as it was closed by #420, and open another issue tracking the problem (I've identified it)

**Comment by yangtaoMark** [comment]
> hmm, ok, but is there noone else using this? Have you tried it yourself?

should be like this :

lua << EOF
local lspconfig = require'lspconfig'
lspconfig.texlab.setup{
  settings = {
    texlab = {
      build = {
        onSave = true;
      },
      forwardSearch = {
        executable = "zathura",
        args = {"--synctex-forward", "%l:1:%f", "%p"},
        onSave = true;
      }
    }
  }
}
EOF
